### ISSUE TITLE
 ARM64 GCStress: Assert failure: obj->GetGCSafeMethodTable ()->HasFinalizer ()

### ISSUE THREAD

**Issue by kyulee1** [dotnet/coreclr#6067]
```
CORECLR! GetCLRRuntimeHost + 0xAEA90 (0x00007ffa`65281a80)
CORECLR! GetCLRRuntimeHost + 0x3962C (0x00007ffa`6520c61c)
CORECLR! GetCLRRuntimeHost + 0x38DB8 (0x00007ffa`6520bda8)
CORECLR! GetCLRRuntimeHost + 0x3991C (0x00007ffa`6520c90c)
CORECLR! GetCLRRuntimeHost + 0x750DEC (0x00007ffa`65923ddc)
CORECLR! GetCLRRuntimeHost + 0x3AAEC0 (0x00007ffa`6557deb0)
CORECLR! GetCLRRuntimeHost + 0x64FF8C (0x00007ffa`65822f7c)
CORECLR! GetCLRRuntimeHost + 0x737564 (0x00007ffa`6590a554)
CORECLR! GetCLRRuntimeHost + 0x7376B4 (0x00007ffa`6590a6a4)
CORECLR! GetCLRRuntimeHost + 0x736474 (0x00007ffa`65909464)
    File: e:\github\coreclr\src\vm\syncblk.cpp Line: 2499
    Image: c:\Users\ddjoin\core_root\CoreRun.exe
```

JIT\Methodical\MDArray\basics\jaggedarr_cs_ro\jaggedarr_cs_ro.cmd 
JIT\Methodical\MDArray\basics\structarr_cs_do\structarr_cs_do.cmd 
JIT\Methodical\MDArray\InnerProd\doublearr_cs_do\doublearr_cs_do.cmd

**Comment by sdmaclea** [comment]
@jashook May be fixed by #9498.

**Comment by BruceForstall** [comment]
Passes under gcstress 0xc
